tools/run_tests/artifacts/artifact_targets.py

```diff
@@ -118,24 +114,27 @@ def pre_build_jobspecs(self):
     def build_jobspec(self):
         environ = {}
         if self.platform == 'linux_extra':
-            # Raspberry Pi build
-            environ['PYTHON'] = '/usr/local/bin/python{}'.format(
+            # Crosscompilation build for armv7 (e.g. Raspberry Pi)
+            environ['PYTHON'] = '/opt/python/{}/bin/python3'.format(
                 self.py_version)
-            environ['PIP'] = '/usr/local/bin/pip{}'.format(self.py_version)
-            # https://github.com/resin-io-projects/armv7hf-debian-qemu/issues/9
-            # A QEMU bug causes submodule update to freeze, so we copy directly
-            environ['RELATIVE_COPY_PATH'] = '.'
-            # Parallel builds are counterproductive in emulated environment
-            environ['GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS'] = '1'
-            extra_args = ' --entrypoint=/usr/bin/qemu-arm-static '
+            environ['PIP'] = '/opt/python/{}/bin/pip3'.format(self.py_version)
+            environ['GRPC_SKIP_PIP_CYTHON_UPGRADE'] = 'TRUE'
+            environ['GRPC_SKIP_TWINE_CHECK'] = 'TRUE'
+            # when crosscompiling, we need to force statically linking libstdc++
+            # otherwise libstdc++ symbols would be too new and the resulting
+            # wheel wouldn't pass the auditwheel check.
+            # This is needed because C core won't build with GCC 4.8 that's
+            # included in the default dockcross toolchain and we needed
+            # to opt into using a slighly newer version of GCC.
+            environ['GRPC_PYTHON_BUILD_WITH_STATIC_LIBSTDCXX'] = 'TRUE'
+
             return create_docker_jobspec(
                 self.name,
-                'tools/dockerfile/grpc_artifact_linux_{}'.format(self.arch),
+                'tools/dockerfile/grpc_artifact_python_linux_{}'.format(
```

Comment:
A cleanup for another time, maybe: We have 10 `"{}".format(...)` calls in this file, and ~20 `"%s" % var` format strings. Do we have a policy to standardize on one or the other?

Reply:
yes, we do have many `%` interpolations, in python3 using `.format` is preferred AFAIK. Agreed that this should be done a cleanup in a separate PR.